llvm/include/llvm/CodeGen/SelectionDAG.h

```diff
@@ -675,10 +675,7 @@ class SelectionDAG {
                       bool isTarget = false, bool isOpaque = false);
 
   SDValue getAllOnesConstant(const SDLoc &DL, EVT VT, bool IsTarget = false,
-                             bool IsOpaque = false) {
-    return getConstant(APInt::getAllOnes(VT.getScalarSizeInBits()), DL, VT,
```

Comment:
Can't this just use -1 and let the implementation extend to APInt if it's larger?

Reply:
Doing that is exactly how most i128 miscompiles get introduced ;) Note that getConstant() performs a zero extension, not a sign extension.

Reply:
But that sounds like a bug, generally constants are sign extended in other contexts in llvm (e.g. in MachineOperand)

Reply:
No, constants in LLVM are zero extended by default. They only get sign extended to 64 bit as a result of normal integer promotion rules. Some APIs have a flag to switch to sign extension instead, and everyone always forgets to set it when necessary, because things look like they work correctly for small integer types. (See also https://github.com/llvm/llvm-project/pull/80309.)